tools/docs/generate2.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""A tool to generate api_docs for TensorFlow Addons.
+
+```
+Run from root folder of source code
+python tools/docs/generate2.py --output_dir=docs/
+```
+
+Requires a local installation of:
+  Docs Generator - pip install git+https://github.com/tensorflow/docs
+  tensorflow_addons
```

Comment:
Yeah we could automate the install/generation with a shell script, but we should be able to build the docs using a bazel without the need to pip install the package first (which would require a nightly build that we do not have yet). 

See the "note" in this section for the reason why they used a pip install instead of just using the source code: https://www.tensorflow.org/community/contribute/documentation#python

Reply:
@seanpmorgan @facaiy : Two questions here:
1. Should I still rely on doc s generator in tensorflow_docs .. or that is not an option ?
2. Should the docs be generated from installed version of addons or from source code version of addons ? if Source code version then I assume it would always be from master branch, Or do we want to keep docs for multiple versions like in TF API docs you can select previous version and see docs generated for previous versions too.

Reply:
To import docs into tensorflow.org, each "subsite" has a config file that specifies the branch for narrative docs and the branch for API docs.
Typically, narrative docs are set to 'master' since it's easier if browsing the GitHub repo looks close to the site (minus changes from last publish date). 
For API docs, we typically publish from a release branch (though it could be a tag). That way you can point to older versions of generated docs in GitHub and the Markdown is still readable. The downside is that you need to bump the (internal) config file whenever you make a release and want to publish it. But we can come up with process for that, or just ping me.